enhancements/local-storage/automatic-detection.md

```diff
@@ -64,40 +65,299 @@ and we can use that information to sort devices into storage classes based on th
+  DiscoveryFailed DiscoveryPhase = "error"
+)
+
+type LocalVolumeDiscoveryStatus struct {
+       // Phase represents the current phase of discovery process
+       // This is used by the OLM UI to provide status information
+       // to the user
+	Phase DiscoveryPhase `json:"phase,omitempty"`
+	// Conditions is a list of conditions and their status.
+	Conditions []operatorv1.OperatorCondition `json:"conditions,omitempty"`
+	// observedGeneration is the last generation change the operator has dealt with
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+type LocalVolumeDiscoverySpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscovery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeDiscoverySpec   `json:"spec"`
+	Status            LocalVolumeDiscoveryStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeDiscoveryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeDiscovery `json:"items"`
+}
+```
+
+API scheme for `LocalVolumeDiscoveryResult` CR:
+```go
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational or nonRotational
+type DeviceMechanicalProperty string
+
+// The mechanical properties of the devices
+const (
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+// DeviceType represents the type of devices that would be discoverd.
+type DeviceType string
+
+// Devices types that should be discovered
+const (
+	// RawDisk represents a device-type of block disk
+	RawDisk DeviceType = "disk"
+	// Loop type device
+	Loop DeviceType = "loop"
+)
+
+type DiscoveredDevice struct {
+	// Name represents the persitent name of the device. For eg, /dev/disk/by-id/...
+	Name string `json:"name"`
+	// Model of the detected device
+	Model string `json:"model,omitempty"`
+	// Type of the detected device
+  Type DeviceType `json:"type"`
+	// Vendor of the detected device
+	Vendor string `json:"vendor,omitempty"`
+	// Size of the detected device
+	Size resource.Quantity `json:"size,omitempty"`
+	// Property represents whether the device type is rotational or not
+	Property DeviceMechanicalProperty `json:"property"`
+        // FSType represents the filesystem available on the device
+	FSType string `json:"fstype",omitempty`
+}
+
+type LocalVolumeDiscoveryResultSpec struct {
+  // NodeName represent the node for which LocalVolumeDiscoveryResult was created
+	NodeName string `json:"nodeName"`
```

Comment:
- We shouldn't have any properties in the result `spec`. All the results should be in the result `status` struct, including the node name. 
- The node name should also be encoded into the name of the CR so that the status for a node can always be queried based on the CR name, which has the node name included. For example, name it `device-status-<node>`. 
   - Because of long node names we may also need to take the hash of the node name, like we do in rook for resources with the node name
   - The node name can also be included as a label to make it easier to query for the results of a node

Reply:
Updated the design to remove the `node` information from `spec` and added it to the `Status`. Also updated the example CR name to - `disovery-result-<hashed-node-name>` and hashed node name label.